website/addons/box/tests/test_views.py

```diff
@@ -0,0 +1,554 @@
+
+    @mock.patch('website.addons.box.views.auth.box_oauth_finish')
+    @mock.patch('website.addons.box.views.auth.finish_auth')
+    @mock.patch('website.addons.box.views.auth.BoxClient')
+    def test_box_oauth_finish(self, mock_get, mock_finish, mock_oauth):
+        mock_client = mock.MagicMock()
+        mock_client.get_user_info.return_value = {'name': 'Mr. Box', 'id': '1234567890'}
+        mock_get.return_value = mock_client
+        mock_finish.return_value = {
+            'token_type': 'something',
+            'access_token': 'something',
+            'refresh_token': 'something'
+        }
+        mock_oauth.return_value = ('mytoken123', 'myboxid', 'done')
+        url = api_url_for('box_oauth_finish')
+
+        res = self.app.get(url)
+
+        assert_is_redirect(res)
+
+    @mock.patch('website.addons.box.model.BoxOAuthSettings.revoke_token')
+    def test_box_oauth_delete_user(self, mock_disable_access_token):
+        self.user.add_addon('box')
+        settings = self.user.get_addon('box')
+        oauth = BoxOAuthSettings(user_id='fa;l', access_token='a;lkjadl;kas')
+        oauth.save()
+        settings.oauth_settings = oauth
+        settings.save()
+        assert_true(settings.has_auth)
+        self.user.save()
+        url = api_url_for('box_oauth_delete_user')
+        self.app.delete(url)
+        settings.reload()
+        assert_false(settings.has_auth)
+
+
+class TestConfigViews(BoxAddonTestCase):
+
+    def setUp(self):
+        super(TestConfigViews, self).setUp()
+        self.user.add_addon('box')
+        settings = self.user.get_addon('box')
+        oauth = BoxOAuthSettings(user_id='not none', _access_token='Nah')
+        oauth.save()
+        settings.oauth_settings = oauth
+        settings.save()
+
+    @mock.patch('website.addons.box.client.BoxClient.get_user_info')
+    def test_box_user_config_get_has_auth_info(self, mock_account_info):
+        mock_account_info.return_value = {'display_name': 'Mr. Box'}
+        url = api_url_for('box_user_config_get')
+        res = self.app.get(url, auth=self.user.auth)
+        assert_equal(res.status_code, 200)
+        # The JSON result
+        result = res.json['result']
+        assert_true(result['userHasAuth'])
+
+    @mock.patch('website.addons.box.client.BoxClient.get_user_info')
+    @mock.patch('website.addons.box.model.BoxNodeSettings.full_folder_path')
+    def test_box_user_config_get_has_valid_credentials(self, mock_data, mock_account_info):
+        mock_account_info.return_value = {'display_name': 'Mr. Box'}
+        url = api_url_for('box_user_config_get')
+        res = self.app.get(url, auth=self.user.auth)
+        assert_equal(res.status_code, 200)
+        # The JSON result
+        result = res.json['result']
+        assert_true(result['validCredentials'])
+
+    @mock.patch('website.addons.box.client.BoxClient.get_user_info')
+    @mock.patch('website.addons.box.model.BoxNodeSettings._fetch_folder_data')
```

Comment:
I made this change, Its easier to mock `_fetch_folder_data` because the return value can just be set and flake8 likes it a bit more.